File: benchmarks/language-comparison/mutations/exhaustive.ts
type Status = "pending" | "done" | "paused";
function label(status: Status): string {
    switch (status) {
        case "pending": return "pending";
        case "done": return "done";
        default: { const unreachable: never = status; return unreachable; }
    }
}
console.log(label("pending"));

File: benchmarks/language-comparison/mutations/exhaustive.rs
enum Status {
    Pending,
    Done,
    Paused,
}
fn label(status: Status) -> &'static str {
    match status {
        Status::Pending => "pending",
        Status::Done => "done",
    }
}
fn main() {
    println!("{}", label(Status::Pending));
}
